src/logging.cpp

```diff
@@ -196,29 +199,31 @@ std::string BCLog::Logger::LogTimestampStr(const std::string &str)
     } else
         strStamped = str;
 
-    if (!str.empty() && str[str.size()-1] == '\n')
-        m_started_new_line = true;
-    else
-        m_started_new_line = false;
-
     return strStamped;
 }
 
 void BCLog::Logger::LogPrintStr(const std::string &str)
 {
-    std::string strTimestamped = LogTimestampStr(str);
+    std::string str_prefixed = str;
+
+    if (m_log_threadnames && m_started_new_line)
+        str_prefixed.insert(0, "[" + util::ThreadGetInternalName() + "] ");
```

Comment:
style-nit: in commit 951769cf12 threads: prefix log messages with thread names

Could add `{}`, according to developer notes for new code?

Reply:
Done, thanks.